OpenSim/Actuators/MuscleFixedWidthPennationModel.h

```diff
@@ -342,19 +325,27 @@ OpenSim_DECLARE_CONCRETE_OBJECT(MuscleFixedWidthPennationModel, Object);
                              double muscleVelocity,
                              double tendonVelocity) const;
 
-    void ensureModelUpToDate();
+protected:
+    // Component interface.
+    void extendFinalizeFromProperties() override;
 
 private:
     void setNull();
     void constructProperties();
-    void buildModel();
-    void ensurePropertiesSet() const;
 
     double m_parallelogramHeight;
     double m_maximumSinPennation;
     double m_minimumFiberLength;
     double m_minimumFiberLengthAlongTendon;
 
+    // Enforces a lower bound on the fiber length to avoid a numerical
+    // singularity as the fiber length approaches zero.
+    double clampFiberLength(double fiberLength) const;
+
+    // These classes are friends because they call clampFiberLength().
+    friend class Thelen2003Muscle;
+    friend class Millard2012EquilibriumMuscle;
+    friend class Millard2012AccelerationMuscle;
```

Comment:
That seems to introduce inconsistencies in fiberLength. If the fiberLength passed in was not within the clamped range to begin with, then does it represent an invalid state? These clamped statements in the Muscle code are troubling since it appears that state variable values are readily computed by the integrator that are out of bounds and then artificially clamped for any state evaluation. Intuitively that seems problematic if dynamic equations become insensitive to trial states beyond some value. E.g. what information can the integrator use to refine the step size if a muscle fiber estimate is ridiculously long but it gets the same derivative value for any l > l_max, for example, because it simply gets clamped to l_max where ever it is used internally to compute a derivative?